tfx/dsl/experimental/spans_resolver.py

```diff
@@ -0,0 +1,158 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Experimental Resolver for getting the latest artifact."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import Dict, Optional, Text
+
+from tfx import types
+from tfx.components.example_gen import utils
+from tfx.dsl.resolvers import base_resolver
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.proto import range_config_pb2
+from tfx.types import artifact_utils
+from tfx.types.standard_artifacts import Examples
+
+
+class SpansResolver(base_resolver.BaseResolver):
+  """Resolver that returns a range of spans in a given Examples channel.
+
+  Note that this Resolver is experimental and is subject to change in terms of
+  both interface and implementation.
+  """
+
+  def __init__(self, 
+               range_config: Optional[range_config_pb2.RangeConfig] = None,
+               merge_same_artifact_type: bool = False):
+    self._range_config = range_config
+    self._merge_same_artifact_type = merge_same_artifact_type
+
+  def resolve(
+      self,
+      pipeline_info: data_types.PipelineInfo,
+      metadata_handler: metadata.Metadata,
+      source_channels: Dict[Text, types.Channel],
+  ) -> base_resolver.ResolveResult:
+    artifacts_dict = {}
+    resolve_state_dict = {}
+
+    # Verifies that duplicate spans are not added to output over multiple
+    # channels; specifically for when `merge_same_artifact_type` is set to True.
+    total_processed_spans = set()
+
+    pipeline_context = metadata_handler.get_pipeline_context(pipeline_info)
+    if pipeline_context is None:
+      raise RuntimeError('Pipeline context absent for %s' % pipeline_context)
+    for k, c in source_channels.items():
```

Comment:
Do we intend to allow multiple channels for this specific resolver? Or we only expect one?

Reply:
currently, we're allowing multiple channels since current plan is to also add ability to resolve multiple channels into one. in the event, we decide to move resolver in different direction, we can add the one channel assertion.

Reply:
re todays  discussion, we will assert only one channel is allowed and resolve that single Examples channel.